test/transcoding/OpenCL/atomic_syncscope_test.ll

```diff
@@ -0,0 +1,158 @@
+; RUN: llvm-as %s -o %t.bc
+; RUN: llvm-spirv %t.bc --spirv-ext=+SPV_EXT_shader_atomic_float_add -o %t.spv
+; RUN: spirv-val %t.spv
+; RUN: llvm-spirv %t.spv -to-text -o %t.spt
+; RUN: FileCheck < %t.spt %s -check-prefix=CHECK-SPIRV
+; RUN: llvm-spirv %t.spv -r --spirv-target-env=CL2.0 -o - | llvm-dis -o %t.rev.ll
+; RUN: FileCheck < %t.rev.ll %s -check-prefix=CHECK-LLVM
+
+target datalayout = "e-p:32:32-i64:64-v16:16-v24:32-v32:32-v48:64-v96:128-v192:256-v256:256-v512:512-v1024:1024"
+target triple = "spir64"
+
+@j = local_unnamed_addr addrspace(1) global i32 0, align 4
+
+; SPIRV scopes
+; 0 - ScopeCrossDevice (all svm devices)
+; 1 - ScopeDevice
+; 2 - ScopeWorkGroup
+; 3 - ScopeSubgroup
+; 4 - ScopeInvocation (mapped from work item)
+
+; OCL scopes
+; 0 - work_item
+; 1 - work_group
+; 2 - device
+; 3 - all_svm_devices
+; 4 - sub_group
+
```

Comment:
Consider testing all scope types (work_item, work_group, ...) for every atomic instruction type (store, load, compare exchange).